drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-backend/src/test/java/org/drools/workbench/screens/globals/backend/server/GlobalsEditorServiceTest.java

```diff
@@ -0,0 +1,105 @@
+import org.guvnor.common.services.backend.util.CommentedOptionFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.workbench.common.services.shared.project.KieProjectService;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.backend.vfs.PathFactory;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.base.options.CommentedOption;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GlobalsEditorServiceTest {
+
+    @Mock
+    private KieProjectService kieProjectService;
+
+    @Mock
+    private IOService ioService;
+
+    @Mock
+    private CommentedOptionFactory commentedOptionFactory;
+
+    @Mock
+    private MetadataServerSideService metadataService;
+
+    private GlobalsEditorService globalsEditorService;
+
+    @Before
+    public void setUp() {
+        globalsEditorService = new GlobalsEditorServiceImpl() {
+            {
+                projectService = GlobalsEditorServiceTest.this.kieProjectService;
+                ioService = GlobalsEditorServiceTest.this.ioService;
+                commentedOptionFactory = GlobalsEditorServiceTest.this.commentedOptionFactory;
+                metadataService = GlobalsEditorServiceTest.this.metadataService;
+            }
+        };
+    }
+
+    @Test
+    public void save() {
+        Path path = PathFactory.newPath( "test",
+                                         "file:///test" );
+        GlobalsModel globalsModel = mock( GlobalsModel.class );
+
+        when( ioService.exists( any( org.uberfire.java.nio.file.Path.class ) ) ).thenReturn( false );
+
+        globalsEditorService.create( path,
+                                     "test",
+                                     globalsModel,
+                                     "comment" );
+
+        verify( ioService,
+                times( 1 ) ).write( any( org.uberfire.java.nio.file.Path.class ),
+                                    anyString(),
+                                    any( CommentedOption.class ) );
+    }
+
+    @Test
+    public void generate() {
+        Path path = PathFactory.newPath( "test",
+                                         "file:///test" );
+        GlobalsModel globalsModel = mock( GlobalsModel.class );
+
+        when( ioService.exists( any( org.uberfire.java.nio.file.Path.class ) ) ).thenReturn( false );
+
+        globalsEditorService.generate( path,
+                                       "test",
+                                       globalsModel,
+                                       "comment" );
+
+        verify( ioService,
+                times( 1 ) ).write( any( org.uberfire.java.nio.file.Path.class ),
+                                    anyString(),
+                                    anyMap(),
```

Comment:
I would probably be nice to capture the ```Map``` and check it has the "generated" flag set to ```true```